person.rb

```diff
@@ -15,6 +15,11 @@ def initialize(age, name = 'Unkown', parent_permission: true)
     @parent_permission = parent_permission
 
     @corrector = Corrector.new
+    @rentals = []
+  end
+
+  def add_rental(person)
+    @rentals.push(person)
   end
```

Comment:
- [ ] What happens when one rents a book? The system must get to record which `person` is renting, the `date` of renting, and the `book` being rented. Therefore it is expected that this method takes two parameters which are, `book` and `date`.

  Another correction. We are not pushing the person object to rentals array(`@rentals.push(person)`), we need to push a rental to the rentals array. Remember the rentals class already takes care of this.
  
  What we should be doing is creating a new Rental in this method. For example:
  
  ```ruby
  def add_rental(book, date)
    # *The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
    Rental.new(date, book, self)
  end
  ```
  Please consider refactoring this method accordingly.